### ISSUE TITLE
Conan config install skips compiler auto detection?

### ISSUE THREAD

**Issue by michaelmaguire** [conan-io/conan#4074]
I have a Jenkins pipeline job which successfully runs conan to build a CMake file.

I invoked it in my Jenkinsfile with:

```
def b = client.run(command: 'create .. PORT/jenkins -s compiler="Visual Studio" -s compiler.version=14 -s compiler.toolset="v140_xp" -s build_type=Debug')
```

I see output as follows:

```
[...]
[windows_build_and_unittest] Running on BNXWNYDNXGJK03 in E:\nX\nX_nightlybuild\workspace\replicant_pipeline
[...]
[windows_build_and_unittest] Auto detecting your dev setup to initialize the default profile (E:\nX\nX_nightlybuild\workspace\replicant_pipeline@tmp\conan.tmp1073750536273988501\.conan\profiles\default)
[windows_build_and_unittest] Found Visual Studio 14
[windows_build_and_unittest] Found gcc 6.4
[windows_build_and_unittest] gcc>=5, using the major as version
[windows_build_and_unittest] Default settings
[windows_build_and_unittest] 	os=Windows
[windows_build_and_unittest] 	os_build=Windows
[windows_build_and_unittest] 	arch=x86_64
[windows_build_and_unittest] 	arch_build=x86_64
[windows_build_and_unittest] 	compiler=Visual Studio
[windows_build_and_unittest] 	compiler.version=14
[windows_build_and_unittest] 	build_type=Release
[windows_build_and_unittest] *** You can change them in E:\nX\nX_nightlybuild\workspace\replicant_pipeline@tmp\conan.tmp1073750536273988501\.conan\profiles\default ***
[windows_build_and_unittest] *** Or override with -s compiler='other' -s ...s***
[windows_build_and_unittest] 
[windows_build_and_unittest] 
[windows_build_and_unittest] Configuration:
[windows_build_and_unittest] [settings]
[windows_build_and_unittest] os=Windows
[windows_build_and_unittest] os_build=Windows
[windows_build_and_unittest] arch=x86_64
[windows_build_and_unittest] arch_build=x86_64
[windows_build_and_unittest] compiler=Visual Studio
[windows_build_and_unittest] compiler.version=14
[windows_build_and_unittest] build_type=Debug
[windows_build_and_unittest] compiler.toolset=v140_xp
[windows_build_and_unittest] [options]
[windows_build_and_unittest] [build_requires]
[windows_build_and_unittest] [env]
[windows_build_and_unittest] 
[...]
[windows_build_and_unittest] -- The C compiler identification is MSVC 19.0.24215.1
[windows_build_and_unittest] -- The CXX compiler identification is MSVC 19.0.24215.1
[windows_build_and_unittest] -- Check for working C compiler: D:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/x86_amd64/cl.exe
[... worky ...]
```

Then, setting up a different pipeline for a different project, I thought I would be smarter and use conan install config so that I could use profiles.  For that project, I invoke from my Jenkinsfile:

```
                    client.run(command: "config install git@internalgithub.com:nx/conan-nx-common-config.git --type git")

[...]
                    client.run(command: 'create . PORT/stable --profile vs2015_14_v140_xp_Debug')
```

But instead of working, what I see is:

```
[...]
Running on BNXWNYDNXGJK03 in E:\nX\nX_nightlybuild\workspace\Conan_pipeline
[...]
[Pipeline] newBuildInfo
[Pipeline] runConanCommand
[Conan_pipeline] $ cmd.exe /C "conan config install git@bbgithub.dev.bloomberg.com:nx/conan-nx-common-config.git --type git && exit %%ERRORLEVEL%%"
Trying to clone repo  git@bbgithub.dev.bloomberg.com:nx/conan-nx-common-config.git
Cloning into 'config'...
Repo cloned
Copying file README.md to E:\nX\nX_nightlybuild\workspace\Conan_pipeline@tmp\conan.tmp6266276910363586376\.conan\.
Copying file registry.txt to E:\nX\nX_nightlybuild\workspace\Conan_pipeline@tmp\conan.tmp6266276910363586376\.conan\.
Installing profiles
    Installing profile default
    Installing profile vs2015_14_v140_xp_Debug
    Installing profile vs2015_14_v140_xp_Release
Processing conan.conf
[Conan_pipeline] $ cmd.exe /C "conan_build_info E:\nX\nX_nightlybuild\workspace\Conan_pipeline@tmp\conan.tmp6266276910363586376\conan_log.log --output E:\nX\nX_nightlybuild\workspace\Conan_pipeline@tmp\conan9198041355346895935build-info && exit %%ERRORLEVEL%%"
[Pipeline] newBuildInfo
[Pipeline] runConanCommand
[...]
[Conan_pipeline] $ cmd.exe /C "conan create . PORT/stable --profile vs2015_14_v140_xp_Debug && exit %%ERRORLEVEL%%"
Exporting package recipe
No linter available. Use a pip installed conan for recipe linting
zlib/1.2.11@PORT/stable export: Copied 1 '.txt' file: CMakeLists.txt
zlib/1.2.11@PORT/stable: A new conanfile.py version was exported
zlib/1.2.11@PORT/stable: Folder: E:\nX\nX_nightlybuild\workspace\Conan_pipeline@tmp\conan.tmp3124279038230407292\.conan\data\zlib\1.2.11\PORT\stable\export
Configuration:
[settings]
os=Windows
os_build=Windows
arch=x86_64
arch_build=x86_64
compiler=Visual Studio
compiler.version=14
compiler.toolset=v140_xp
build_type=Debug
[options]
[build_requires]
[env]

zlib/1.2.11@PORT/stable: WARN: Forced build from source
zlib/1.2.11@PORT/stable (test package): Installing E:\nX\nX_nightlybuild\workspace\Conan_pipeline\test_package\conanfile.py
Requirements
    zlib/1.2.11@PORT/stable from local cache - Cache
Packages
    zlib/1.2.11@PORT/stable:d5817f13e39c3c4195e51ee5243d672b6c26d94b - Build

zlib/1.2.11@PORT/stable: Building your package in E:\nX\nX_nightlybuild\workspace\Conan_pipeline@tmp\conan.tmp3124279038230407292\.conan\data\zlib\1.2.11\PORT\stable\build\d5817f13e39c3c4195e51ee5243d672b6c26d94b
zlib/1.2.11@PORT/stable: Configuring sources in E:\nX\nX_nightlybuild\workspace\Conan_pipeline@tmp\conan.tmp3124279038230407292\.conan\data\zlib\1.2.11\PORT\stable\source
C:\Program Files\Conan\conan\urllib3\connectionpool.py:858: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings

zlib/1.2.11@PORT/stable: Checking that zip sha256_hash matches [c3e5e9fdd5004dcb542feda5ee4f0ff0744628baf8ed2dd5d66f8ca1197cb1a1]
zlib/1.2.11@PORT/stable: Copying sources to build folder
zlib/1.2.11@PORT/stable: Generator cmake created conanbuildinfo.cmake
zlib/1.2.11@PORT/stable: Calling build()
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:2 (PROJECT):
  No CMAKE_C_COMPILER could be found.
[... failure ...]
```

It seems like it skipped the ```Auto detecting your dev setup to initialize the default profile``` step because I installed config?

To help us debug your issue please explain:

- [x] I've read the [CONTRIBUTING guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [1.8.4] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

**Comment by memsharded** [comment]
Hi!

I have tried to reproduce with:
- Removing my local cache to make sure no auto-detect is there
- Creating a profile with your configuration above in the local cache
- ``conan install zlib/1.2.11@conan/stable --build=zlib -pr=myprofile``

It seems that it is able to build without problems. Did you do any modification to your zlib/1.2.11@PORT/stable recipe, with respect to the conan public one? Any change to the conanfile, (specially the build() method), or to the CMakeLists.txt that you could share? 


Also beware that you might be distributing unwanted things in your ``conan config install``. I see:

```
Copying file registry.txt to E:\nX\nX_nightlybuild\workspace\Conan_pipeline@tmp\conan.tmp6266276910363586376\.conan\.
```

The registry.txt shouldn't be redistributed. If you want to configure remotes, put a ``remotes.txt`` file there.

**Comment by michaelmaguire** [comment]
I had it called ```remotes.txt``` at first, then I got errors:

```
[Pipeline] newBuildInfo
[Pipeline] runConanCommand
[Conan_pipeline] $ cmd.exe /C "conan config install git@bbgithub.dev.bloomberg.com:nx/conan-nx-common-config.git --type git && exit %%ERRORLEVEL%%"
Trying to clone repo  git@internalgithub.com:nx/conan-nx-common-config.git
Cloning into 'config'...
Repo cloned
Copying file README.md to E:\nX\nX_nightlybuild\workspace\Conan_pipeline@tmp\conan.tmp2827515684683227565\.conan\.
Defining remotes
WARN: Remotes registry file missing, creating default one in E:\nX\nX_nightlybuild\workspace\Conan_pipeline@tmp\conan.tmp2827515684683227565\.conan\registry.txt
Installing profiles
    Installing profile default
    Installing profile vs2015_14_v140_xp_Debug
    Installing profile vs2015_14_v140_xp_Release
Processing conan.conf
[Conan_pipeline] $ cmd.exe /C "conan_build_info E:\nX\nX_nightlybuild\workspace\Conan_pipeline@tmp\conan.tmp2827515684683227565\conan_log.log --output E:\nX\nX_nightlybuild\workspace\Conan_pipeline@tmp\conan8056729440755303049build-info && exit %%ERRORLEVEL%%"
[Pipeline] newBuildInfo
[Pipeline] runConanCommand
```

**Comment by memsharded** [comment]
Not easy to know...

I think I'd try to rule out any possible interaction of the ``conan config install`` and the creation of the default profile, by doing:
- Putting the ``vs2015_14_v140_xp_Debug`` profile, in the repo itself.
- Run ``create . PORT/stable --profile=./vs2015_14_v140_xp_Debug`` (the profile is local, not installed)

Please let me know if you can run that and if it works.
A second iteration if that works is to put the profile in the repo of the ``conan config install`` and make sure that it is the only thing there, no any other file in the ``conan config install`` source repo.

**Comment by michaelmaguire** [comment]
I apologize -- I've confirmed the problem isn't with conan config loading. 

I tried building the same project that was building before ('replicant' which arguably is more complex and depends on zlib) rather than 'zlib' and my build zlib seems to be the issue.

So something about my attempt to conan create  zlib on my build server is failing, where it succeeds on my personal dev server.  

I'll do some more investigation.  Thanks for the response and sorry for the false alarm.
